tests/all/component_model/instance.rs

```diff
@@ -55,3 +55,20 @@ fn instance_exports() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn instantiate_with_type_exports() -> Result<()> {
+    // For now, this just tests that we can instantiate a component with type exports
+    let engine = super::engine();
+    let component = r#"
+        (component
+            (type string)
+            (export "" (type 0))
+        )
```

Comment:
I think this can also be added to something in `tests/misc_testsuite/component-model` if you'd prefer

Reply:
I'll move it there, thanks!

Reply:
Are we running `misc_testsuite` with `component-model` feature flag in CI? A quick perusal of `main.yml` seems like we're only running the `component_model` test with `cargo test --test all --features component-model component_model`.

Reply:
Never mind, that does indeed run both sets as expected.